### ISSUE TITLE
@emotion/serialize/types/index.ts TS2411: error

### ISSUE THREAD

**Issue by ovaris** [emotion-js/emotion#3136]
**Current behavior:**

TypeScript compiler throws numerous errors related to `CSSInterpolation`:
```
node_modules/@emotion/serialize/types/index.d.ts:37:18 - error TS2411: Property 'zoom' of type 'Zoom | readonly NonNullable<Zoom | undefined>[] | ("normal" | "reset" | (string & {}) | Globals)[] | undefined' is not assignable to 'string' index type 'CSSInterpolation'.

Found 825 errors in the same file, starting at: node_modules/@emotion/serialize/types/index.d.ts:37
```

Using TS version 5.3.2

Not sure what changed but suddenly index.d.ts appered under node_modules/@emotion/serialize/types even though we have been using same version 1.1.2 all the time (or actually it's dependency of react-select)

└─┬ react-select@5.7.2
    └─┬ @emotion/react@11.11.1
      ├─┬ @emotion/babel-plugin@11.11.0
      │ └── @emotion/serialize@1.1.2 deduped
      └── @emotion/serialize@1.1.2

**Comment by mitchhentgesspotify** [comment]
Hey, I'm seeing the same thing! I've got a [repro repo over here for you, Andarist](https://github.com/mitchhentgesspotify/emotion-tsc)

**Comment by kamenminkovcom** [comment]
This is causing the issue - https://github.com/frenic/csstype/issues/189

**Comment by sg785-sakshi** [comment]
> 
> 
> **Current behavior:**
> 
> TypeScript compiler throws numerous errors related to `CSSInterpolation`:
> ```
> node_modules/@emotion/serialize/types/index.d.ts:37:18 - error TS2411: Property 'zoom' of type 'Zoom | readonly NonNullable<Zoom | undefined>[] | ("normal" | "reset" | (string & {}) | Globals)[] | undefined' is not assignable to 'string' index type 'CSSInterpolation'.
> 
> Found 825 errors in the same file, starting at: node_modules/@emotion/serialize/types/index.d.ts:37
> ```
> 
> Using TS version 5.3.2
> 
> Not sure what changed but suddenly index.d.ts appered under node_modules/@emotion/serialize/types even though we have been using same version 1.1.2 all the time (or actually it's dependency of react-select)
> 
> └─┬ react-select@5.7.2
>     └─┬ @emotion/react@11.11.1
>       ├─┬ @emotion/babel-plugin@11.11.0
>       │ └── @emotion/serialize@1.1.2 deduped
>       └── @emotion/serialize@1.1.2
> 
> 

Having the same issue as well

**Comment by heath-freenome** [comment]
I'm seeing the same thing in my builds now too. Any idea when the fix will be released?

**Comment by therockerline** [comment]
I have the same error in my repository, I saw a pull request for csstype in range, when is it accepted?

**Comment by suren-atoyan** [comment]
I have the same issue!

**Comment by w9nder** [comment]
Same issue here.

**Comment by heegu0311** [comment]
Same issue here.

**Comment by kauan-carvalho-code** [comment]
I was experiencing the same issue, and it was resolved by setting the skipLibCheck property to true within the compilerOptions in the tsconfig file.

**Comment by ERICWILFER** [comment]
same issue here

**Comment by heegu0311** [comment]
@kauan-carvalho-code Thanks. In my case, my IDE(webstorm) works too slowly to keep coding.
It's probably due to so many errors(more than  800). After changing skipLibCheck to true, IDE is working fine now.

**Comment by yanachrnsh** [comment]
same issue

**Comment by heath-freenome** [comment]
> @kauan-carvalho-code Thanks. In my case, my IDE(webstorm) worked too slowly to keep coding. It was probably due to so many errors(more than 800). After changing skipLibCheck to true, IDE is working fine now. 😃

I tried doing that for my builds and it still fails

**Comment by sacummings91** [comment]
I'm pretty sure it's cuz `skipLibCheck: true` doesn't solve the issue if you are importing types into your application code.

**Comment by sacummings91** [comment]
If anyone is looking for a workaround I used the solution in this comment pinning csstype to 3.1.2 for now. https://github.com/frenic/csstype/issues/189#issuecomment-1853012588

**Comment by emmatown** [comment]
Fixed by #3141
